eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -21,3 +21,195 @@ function Get-NamepspacesFromDll($dllPath) {
     }
     return $namespaces | Sort-Object -Unique
 }
+
+function DownloadNugetPackage($package, $version, $destination) {
+    # $PackageSourceOverride is a global variable provided in
+    # Update-DocsMsToc.ps1. Its value can set a "customSource" property.
+    # If it is empty then the property is not overridden.
+    $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+    $defaultDownloadUri = "https://www.nuget.org/api/v2/package/$package/$version"
+    try {
+        if ($customPackageSource) {
+            # Download package from devop public feeds
+            Write-Host "Download from public feeds: $customPackageSource"
+            # Invoke the download link and store it into destination
+            nuget install -Source $customPackageSource $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+        }
+        else {
+            Write-Host "Download from nuget: $defaultDownloadUri"
+            # Invoke the download link and store it into destination
+            nuget install $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination # -Source $defaultDownloadUri 
+        }
+    }
+    catch {
+        Write-Error $_ -ErrorAction Continue
+        Write-Host "Failed to download. Try again using default uri: $defaultDownloadUri"
+        # Add fallback logic
+        nuget install $package -Version $version -DirectDownload -DependencyVersion Ignore -OutputDirectory $destination
+    }
+}
+
+function Fetch-NamespacesFromNupkg ($package, $version) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "extractNupkg/$package-$version")
+    if (!(Test-Path $tempLocation)) {
+        New-Item -ItemType Directory -Path $tempLocation -Force | Out-Null
+    }
+    $nupkgFilePath = "$tempLocation/$package-$version.nupkg"
+    Write-Host "Downloading nupkg packge to $nupkgFilePath ...."
+    DownloadNugetPackage -package $package -version $version -destination $nupkgFilePath
+    if (!(Test-Path $nupkgFilePath)) {
```

Comment:
Part of the value of the nuget directdownload is you also don't need to extract the package as it does it for you already.